### PR TITLE
support insertion of custom xml in Pay Now SAML payloads

### PR DESCRIPTION
**Pull request by mdkaraman**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

# What is the ticket # detailing the issue?

Ticket: IVL-184462876

# A brief description of the changes

Current behavior:
Pay Now SAML generator does not have the ability to embed custom xml payloads.

New behavior:
Pay Now SAML generator can call external operations that generate custom xml payloads to insert in the Additional Information attribute.

# Environment Variable

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. Please share the name of the variable below that would enable/disable the feature and which client it applies to.

Variable name: CAREFIRST_PAY_NOW_EMBED_XML_IS_ENABLED

- [x] DC
- [ ] ME

### CODE REVIEW

**Review by erubinst — DISMISSED** [review]
Looks good to me - new structure for finding xml operation folder for specific carrier easy to follow.

File: app/domain/operations/generate_saml_response.rb [hunk 1]
```diff
@@ -36,8 +36,8 @@ def create_payment_transaction(hbx_enrollment, source)
       payment ? Success(payment) : Failure("Issue with Payment transcation")
     end
 
-    def init_saml_generator(payment_transaction, hbx_enrollment)
-      saml_obj = OneLogin::RubySaml::SamlGenerator.new(payment_transaction.payment_transaction_id, hbx_enrollment)
+    def init_saml_generator(payment_transaction, hbx_enrollment, pay_now_key)
```
Comment (scaustin34) [inline app/domain/operations/generate_saml_response.rb:39]: might want to clarify which key? pay_now_carrier_key, perhaps? not critical just a thought. At a glance i wasnt sure what is being passed around

**Review by scaustin34 — DISMISSED** [review]

File: app/domain/operations/generate_saml_response.rb (hunk 1 above)
Comment (mdkaraman) [reply app/domain/operations/generate_saml_response.rb]: Thanks for commenting on this.  Got me to reconsider my previous refactor and also the existing Pay Now code, which led me to the realization that we don't need to pass the key around at all.  New approach requires fewer changes and should be more readable.  Let me know what you think!

**Review by scaustin34 — APPROVED** [review]
Elegant refactor solution, i dig it. LGTM!

**Review by erubinst — APPROVED** [review]

**Pull request merged**